lineman/app/services/thread_query_service.coffee

```diff
@@ -52,15 +43,19 @@ angular.module('loomioApp').factory 'ThreadQueryService', (Records) ->
       parts = options.split ' '
       moment().startOf('day').subtract(parseInt(parts[0]), parts[1])
 
-    createFilter = (filter = 'show_all') ->
-      (viewData) ->
-        _.filter viewData, (thread) ->
-          return false if thread.isMuted() and filter != 'show_muted'
-          return false if thread.readerNotLoaded()
-          switch filter
-            when 'show_all'           then true
-            when 'show_muted'         then thread.isMuted()
-            when 'show_unread'        then thread.isUnread()
-            when 'show_participating' then thread.isParticipating()
-            when 'show_starred'       then thread.isStarred()
-            when 'show_proposals'     then thread.hasActiveProposal()
+    applyFilters = (view, filters, queryType) ->
```

Comment:
This looks really great. 

Have you considered using applyFind on the simple properties? I think using Find that means Loki can use indices rather than apply a function to every record.